### ISSUE TITLE
vSphere CPI pods will never schedule on RKE2 due to nodeSelector in Helm chart

### ISSUE THREAD

**Issue by mstrent** [kubernetes/cloud-provider-vsphere#638]
https://github.com/kubernetes/cloud-provider-vsphere/blob/1650759df957b8d22d4f2fa32fc331e250163fe5/charts/vsphere-cpi/templates/daemonset.yaml#L43

The nodeSelector in the provided vSphere CPI Helm chart will result in pods never scheduling on RKE2 clusters.

Instead, the nodeSelector should be: `node-role.kubernetes.io/control-plane: "true"`

Adding `--set-string daemonset.nodeSelector."node-role\.kubernetes\.io/control-plane"=true` to the Helm command line will work around the issue.

**Comment by lubronzhan** [comment]
Hi @mstrent do you know what tool does RKE use to spin up k8s bits?
Currently, CPI supports Kubeadm installed cluster by default. It labels control plane node with `node-role.kubernetes.io/control-plane=""` according to the doc here:
https://kubernetes.io/docs/reference/setup-tools/kubeadm/implementation-details/#mark-the-node-as-control-plane
That's why we only has the label `node-role.kubernetes.io/control-plane=""`
But you can also change the node selector by changing the values.yaml and the way you mentioned. But But for convenience, let me add both label to the helm chart template them.
BTW, do you know if the taint of RKE cluster control plane node is also different?

**Comment by lubronzhan** [comment]
I tried, the nodeSelector can only have one value. But looks like only RKE user is using it, let me change it to `node-role.kubernetes.io/control-plane="true"`

**Comment by lubronzhan** [comment]
Ok I fixed the chart. But to keep the release version consistent, I release the 1.24.0. 
You need to 
```
helm repo remove vsphere-cpi
helm repo add vsphere-cpi https://kubernetes.github.io/cloud-provider-vsphere
```
Then install it

**Comment by lubronzhan** [comment]
Or let me release a 1.24.1 tonight

**Comment by lubronzhan** [comment]
I just released 1.24.1. Feel free to upgrade it

**Comment by mstrent** [comment]
Thank you!
